### ISSUE TITLE
Energy drift in NVE MD with potential generated with "se_atten" descriptor

### ISSUE THREAD

**Issue by tella94** [deepmodeling/deepmd-kit#2811]
### Summary

Good Afternoon, 

when I train my model using the hybrid descriptors "se_e2_a" and "se_e3" the training goes perfectly fine and in the end i obtain a RMSE on energy of 7 meV/atom while on forces of 0.13 eV/ang. 

However if I substitute the descriptor "se_e2_a" with "se_atten", even if the final RMSE reduces to 4 meV/atom and 0.10 meV and the structural properties of the model are better represented, when I perform NVE simulation with the Lammps code (but I also tried ASE) the total energy drifts upwards. The drift is on the order of 10 meV/atom over 100 ps. Reducing the timestep does not reduce the drift. The curious thing is that the more I increase the number of attention layers, the more this drift becomes pronounced. 

Using "se_atten_v2" does not change anything.  Switching the keyword             "stripped_type_embedding" to true seems to reduce a little the drift but I have to test more. The database of the two different training are identical. Below I attach the two different part of input.json

### DeePMD-kit Version

2.2.3

### TensorFlow Version

2.13

### Python Version, CUDA Version, GCC Version, LAMMPS Version, etc

Python 3.10.8, Cuda 11.8, GCC 11.3.0 , Lammps 22Jun2022

### Details

Train 1:

        "descriptor" :{
            "type":             "hybrid",
            "list":  [
            {
                "type":             "se_e2_a",
                "sel":              [80,40],
                "rcut_smth":        2.00,
                "rcut":             7.00,
                "type_one_side":    false,
                "neuron":           [25,50,100],
                "resnet_dt":        true,
                "axis_neuron":      35,
                "seed":             1
            },
            {
                "type":             "se_e3",
                "sel":              [20,15],
                "rcut_smth":        2.00,
                "rcut":             4.00,
                "neuron":           [5,10,20],
                "resnet_dt":        true,
                "seed":             1
            }

            ]
        },

        "fitting_net" : {
            "neuron":           [240,240,240],
            "resnet_dt":        true,
            "atom_ener": [-102.29717333422299,-218.98450580235118],
            "seed":             1
        }

Train 2:

           "type":             "hybrid",
            "list":  [
            {
                "type":             "se_atten",
                "sel":              90,
                "rcut_smth":        2.00,
                "rcut":             7.00,
                "neuron":           [25,50,100],
                "resnet_dt":        true,
                "axis_neuron":      40,
                "attn":         128,
                "attn_layer":   3,
                "attn_dotr":    true,
                "attn_mask":    false,
                "seed":             1
            },
            {
                "type":             "se_e3",
                "sel":              [20,15],
                "rcut_smth":        2.00,
                "rcut":             4.00,
                "neuron":           [5,10,20],
                "resnet_dt":        true,
                "seed":             1
            }

            ]
        },

        "fitting_net" : {
            "neuron":           [240,240,240],
            "resnet_dt":        true,
            "seed":             1

**Comment by tella94** [comment]
Good Morning, 
I tried to reproduce this behaviour with your AlMgCu system that you uploaded in your google drive. I used the same "se_atten" input script and training and validation set. I generated the potential stopping at the 5 milion step out of 16 milion, so these are preliminary results. However RMSE on forces and energies are quite low (43 meV/Ang on forces and 5 me/atom on energies). 

I used the potential on a 300-atom cubic model (21 Ang of edge size, don't know if this is the right density) of Mg67Al33 generated in the liquid and quenched to 400 K. Then I run NVE starting from this point. Also with this potential there is energy drift, even if it is quite lower than mine (5 meV/atom in 200 ps vs 10 meV/atom in 100 ps that is mine), here I attach the total energy vs time curve: Could you please reproduce this behaviour? Maybe I could have installed lammps in a wrong manner so there are truncation errors in the calculation of the forces.

Thank you very much for your response

[MgAl.pdf](https://github.com/deepmodeling/deepmd-kit/files/12683721/MgAl.pdf)

**Comment by wanghan-iapcm** [comment]
Dear Tella 

Thank you for reporting the issue. Could you please check with your training system that the `sel` is large enough for a cut-off radius of 7? 

We will also try to reproduce the energy drift in the almgcu system. Could you please provide the training input script for the system? 

Thank you in advance

**Comment by tella94** [comment]
Dear Han,
Yes, I checked sel, at 7 Ang the maximum number of neighbors is 79, calculated by deepmd neighbor-stat. However, I tried to increase sel at 120 in another training just to be sure and there is still drift (even if I stopped at the first training steps because with that sel the training is very computationally demanding).

I tried also to increase the cutoff to 9 Ang and sel to 160 (the max number of neighbors at that cutoff) and the energy drift "seems" to reduce. However, also in this case I stopped at the first training steps, so this behaviour is not reliable I think. I'm sorry but my computational resources are limited so I cannot test much in a short time. Here I attach my full script (sorry but since this page is public I have hidden the atom types)

{
    "model": {
	"type_map":	["*","*"],

      "descriptor" :{
            "type":             "hybrid",
            "list":  [
            {
                "type":             "se_atten",
                "sel":              90,
                "rcut_smth":        2.00,
                "rcut":             7.00,
                "neuron":           [25,50,100],
                "resnet_dt":        true,
                "axis_neuron":      40,
		"attn":         128,
                "attn_layer":   3,
                "attn_dotr":    true,
                "attn_mask":    false,
                "seed":             1
            },
            {
                "type":             "se_e3",
                "sel":              [20,15],
                "rcut_smth":        2.00,
                "rcut":             4.00,
                "neuron":           [5,10,20],
                "resnet_dt":        true,
                "seed":             1
            }

            ]
        },



        "fitting_net" : {
            "neuron":           [240,240,240],
	    "resnet_dt":        true,
            "seed":             1
        }

    },
    "learning_rate" :{
	"type":		"exp",
	"decay_steps":   14600,
	"start_lr":	1.0e-3,
	"stop_lr":	3.51e-8
    },

    "loss" :{
	"start_pref_e":	1,
	"limit_pref_e":	100,
	"start_pref_f":	8000,
	"limit_pref_f":	80,
	"start_pref_v":	1,
	"limit_pref_v":	40
    },

    "training" : {
        "training_data"  :{	    
             "systems":["./*_144","./*_108","./*_100","./*_Strano","./*_Strano_2","./*_Cryst","./*_Cryst"],
             "set_prefix":   "set",
             "sys_probs": [0.19, 0.3, 0.35, 0.005, 0.005, 0.10, 0.05],
	     "batch_size":   10
	},
	
	"validation_data"  :{
             "systems":["./Test_*_144","./Test_*_108","./Test_*_100","./Test_*_Strano","./Test_*_2","./Test_*_Cryst","./Test_*_Cryst"],
             "set_prefix":   "set",
             "sys_probs": [0.19, 0.3, 0.35, 0.005, 0.005, 0.10, 0.05],
	     "batch_size":  20,
	     "numb_batch":   7
	},

	"disp_file":	"lcurve.out",
	"disp_freq":	  730,
        "numb_steps":  730000,
	"save_freq":      730,
	"save_ckpt":	"model.ckpt",
	"disp_training":    true,
	"time_training":    true,
	"profiling":      false,
	"profiling_file":  "timeline.json"
    }
}

Thank you very much for your support

**Comment by njzjz** [comment]
> Yes, I checked sel, at 7 Ang the maximum number of neighbors is 79, calculated by deepmd neighbor-stat.

I suggest checking the trajectory instead of the training data.

**Comment by tella94** [comment]
I checked now. In the NVE trajectory where the total energy drifts up the maximum number of neighbors is 65, calculated with a 19200-atom model (the 79 max neighbors that I obtained from the training data is an overestimation due to system that is much densier in comparison with the system in which the energy drifts). I must add that when during the NVE run the system does not become unphysical (no atoms getting too closer or doing strange things). The only thing that happens is that temperature and potential energy increase and hence the structure becomes the one expected at that temperature.

**Comment by tella94** [comment]
Hi, do you have any update on this issue?

Thank you

**Comment by wanghan-iapcm** [comment]
Dear Tella, 
we have reproduced the issue. We will come back to you as soon as we find the solution. 
Best, Han

**Comment by wanghan-iapcm** [comment]
@tella94 you may also access the fixed code by [v2.2.6 ](https://github.com/deepmodeling/deepmd-kit/releases/tag/v2.2.6)

**Comment by tella94** [comment]
Dear Han,
Thank you very much for your support!
